data/codec/src/equivalent/codec_convert.rs

```diff
@@ -9,9 +9,14 @@ where
     Medium: Default + TopDecodeMultiInput + TopEncodeMultiOutput,
 {
     let mut medium: Medium = Default::default();
-    let Ok(()) = from.multi_encode_or_handle_err(&mut medium, PanicErrorHandler);
-    let Ok(result) = To::multi_decode_or_handle_err(&mut medium, PanicErrorHandler);
-    result
+    match from.multi_encode_or_handle_err(&mut medium, PanicErrorHandler) {
+        Ok(()) => (),
+        Err(err) => panic!("panic occured: {:#?}", err),
```

Comment:
`unreachable!()` instead of panic

Also: we should provide a function or trait with one method to extract the data out of Result<T, Infallible>, and we should reuse that. There is too much duplicated code.

Something like ".unwrap_infallible()".

".unwrap()" is also fine, but maybe having a separate name makes it easier for us to revert, once the exhaustive patterns get stabilized.

Reply:
Somebody else thought about this. Import, or just copy code
https://docs.rs/unwrap-infallible/latest/src/unwrap_infallible/lib.rs.html#1-99